pelican/tests/test_readers.py

```diff
@@ -22,6 +22,50 @@ def read_file(self, path, **kwargs):
         r = readers.Readers(settings=get_settings(**kwargs))
         return r.read_file(base_path=CONTENT_PATH, path=path)
 
+    def assertDictHasSubset(self, dictionary, subset):
+        for key, value in subset.items():
+            if key in dictionary:
+                real_value = dictionary.get(key)
+                self.assertEqual(
+                    value,
+                    real_value,
+                    "Expected %r to have value %r, but was %r" % (key, value, real_value))
+            else:
+                self.fail('Expected ' + repr(key) + ' to have value ' +
```

Comment:
It'd be better if this also used string formatting like above (with `str` wrapper as well).